scripts/builtin/mice_linearReg.dml

```diff
@@ -0,0 +1,147 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Assumptions
+# 1. The data is continous/numerical 
+# 2. The missing values are denoted by zeros
+
+# Builtin function Implements binary-class SVM with squared slack variables
+#
+# INPUT PARAMETERS:
+# ---------------------------------------------------------------------------------------------
+# NAME            TYPE    DEFAULT     MEANING
+# ---------------------------------------------------------------------------------------------
+# X               Double   ---         matrix X of feature vectors
+# iter            Integer    3         Number of iteration for multiple imputations 
+# complete        Integer    3         A complete dataset generated though a specific iteration
+# ---------------------------------------------------------------------------------------------
+ 
+
+#Output(s)
+# ---------------------------------------------------------------------------------------------
+# NAME                  TYPE    DEFAULT     MEANING
+# ---------------------------------------------------------------------------------------------
+# dataset               Double   ---        imputed dataset
+# singleSet             Double   ---        A complete dataset generated though a specific iteration
+
+
+
+m_mice_linearReg = function(Matrix[Double] X, Integer iter = 3, Integer complete = 3)
+return(Matrix[Double] dataset, Matrix[Double] singleSet)
+  {
+  n = nrow(X)
+  row = n*complete;
+  col = ncol(X);
+  R = matrix(0, rows=1, cols=col)
+  MR = matrix(0, rows=1, cols=col)
+  
+  mu = colMeans(X) #mean of columns for initial imputation
+
+# storing the mask/address of missing values
+  Mask = X==0;
+  Mask_Filled = Mask;
+
+# filling the missing data with thier means
+  X2 = X+(Mask*mu)
+  
+#slicing non-missing dataset for training columnwise linear regression
+  inverseMask = 1 - Mask;
+
+  for(k in 1:iter)
+  {
+    for(i in 1:col)
+    {
+	  slice_X = removeEmpty(target=X2, margin = "rows", select = inverseMask[,i])
+	  train_Y = slice_X[,i]
+	  
+	  if(i == 1)
+	  {
+	    train_X = slice_X[, (i+1): col]
+	  }
+	  else if (i == col)
+	  {
+	    train_X= slice_X[, 1:col-1];
+	  }
+	  else{
+	    train_X= cbind(slice_X[, 1:i-1], slice_X[,i+1:col]);
+	  }
+	  
+	  
+# slicing dataset for filling missing values
+	  slice_Y = removeEmpty(target=X2, margin = "rows", select = Mask[,i])
+	  test_Y = slice_Y[,i]
```

Comment:
It's a good idea to indent the comment to the same level of the block of code on which it is applied.